// react-native-color-matrix-image-filters/MatrixFilterConstructor/out/Image.js
import { setType } from "./fable-core/Symbol";
import _Symbol from "./fable-core/Symbol";
import { equals as equals_1, equalsRecords, Interface, Option } from "./fable-core/Util";
import { printf, toText } from "./fable-core/String";
import { imageHeight } from "./Constants";

class Model_ {
  constructor(name, source) {
    this.Name = name;
    this.Source = source;
  }

  [_Symbol.reflection]() {
    return {
      type: "MatrixFilterConstructor.Image.Model'",
      interfaces: ["FSharpRecord", "System.IEquatable"],
      properties: {
        Name: "string",
        Source: Option(Interface("Fable.Helpers.ReactNative.Props.IImageSource"))
      }
    };
  }

  Equals(other) {
    return equalsRecords(this, other);
  }

}

export { Model_ as Model$27$ };
setType("MatrixFilterConstructor.Image.Model'", Model_);
export class Model {
  constructor(tag, data) {
    this.tag = tag | 0;
    this.data = data;
  }

  [_Symbol.reflection]() {
    return {
      type: "MatrixFilterConstructor.Image.Model",
      interfaces: ["FSharpUnion", "System.IEquatable"],
      cases: [["Concrete", Model_], ["Random", Model_], ["FromPicker", Model_]]
    };
  }

  Equals(other) {
    return this === other || this.tag === other.tag && equals_1(this.data, other.data);
  }

}
setType("MatrixFilterConstructor.Image.Model", Model);
export const defaultImage = new Model(0, new Model_("Parrot", require("../parrot.png")));
export function random() {
  const id = Math.round(Math.random() * 992);
  const timestamp = Date.now();
  const uri = toText(printf("https://picsum.photos/%f?image=%f&t=%f"))(imageHeight, id, timestamp);
  return new Model(1, new Model_("Random", {
    uri: uri
  }));
}
export function fromPicker(source) {
  return new Model(2, new Model_("Pick image", source));
}
export const availableImages = [defaultImage, new Model(0, new Model_("React logo", {
  uri: "https://tinyurl.com/y8xs3ehd"
})), new Model(0, new Model_("Triangle", {
  uri: "https://tinyurl.com/ycedtewy"
})), random(), fromPicker(null)];
export function equals(first, second, _arg1_0, _arg1_1) {
  const _arg1 = [_arg1_0, _arg1_1];
  const $var1 = _arg1[0].tag === 1 ? _arg1[1].tag === 1 ? [0] : [2] : _arg1[0].tag === 2 ? _arg1[1].tag === 2 ? [1] : [2] : [2];

  switch ($var1[0]) {
    case 0:
      return true;

    case 1:
      return true;

    case 2:
      return equals_1(first, second);
  }
}
export function source(_arg1) {
  const $var2 = _arg1.tag === 1 ? [0, _arg1.data] : _arg1.tag === 2 ? [0, _arg1.data] : [0, _arg1.data];

  switch ($var2[0]) {
    case 0:
      return $var2[1].Source;
  }
}
export function name(_arg1) {
  const $var3 = _arg1.tag === 1 ? [0, _arg1.data] : _arg1.tag === 2 ? [0, _arg1.data] : [0, _arg1.data];

  switch ($var3[0]) {
    case 0:
      return $var3[1].Name;
  }
}
